pixeltable/globals.py

```diff
@@ -600,18 +601,32 @@ def create_dir(
         Create a directory and replace if it already exists:
 
         >>> pxt.create_dir('my_dir', if_exists='replace_force')
+
+        Create a subdirectory along with its ancestors:
+
+        >>> pxt.create_dir('parent1.parent2.sub_dir', create_parents=True)
     """
     path_obj = catalog.Path(path)
     cat = Catalog.get()
 
     with env.Env.get().begin_xact():
+        if create_parents:
+            # start walking from the root
+            ancestors = path_obj.ancestors()
+            root_path = catalog.Path('', empty_is_valid=True)
```

Comment:
If ancestors() included the root, you wouldn't need this special-casing.

Reply:
let me know what you think we should do when ancestors() called on root. for this use case its simple but for any other uses of walking down from root perhaps we can none (which means users of ancestors need to handle it)